### DISCUSSION TITLE
select not working with relation

### DISCUSSION THREAD

**Discussion by shankhadevpadam** [filamentphp/filament#12868] [Help]
### Package

Panel builder

### Package Version

v3.2.80

### How can we help you?

Why this select occur error. I used spatie permission package for role and permission.
```
Forms\Components\Select::make('roles')
    ->relationship('roles', 'name')
    ->getOptionLabelFromRecordUsing(fn (Model $record) => Str::headline($record->name))
    ->required(),
```

Output

```
array_diff(): Argument #2 must be of type array, string given
```

**Comment by leandrocfe** [comment]
Does it work without `getOptionLabelFromRecordUsing` ?

**Reply by shankhadevpadam** [reply]
Yes, it works. If I removed getOptionLabelFromRecordUsing

**Reply by shankhadevpadam** [reply]
```
Forms\Components\Select::make('roles')
    ->relationship('roles', 'name')
    ->required(),
```

This also triggers same issue.

**Reply by leandrocfe** [reply]
Are you using Filament shield or spatie roles/permissions?
Should this select be `multiple()`?

**Reply by shankhadevpadam** [reply]
Yes, I am using a filament shield. Single select option.
